source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -232,99 +232,53 @@ void MaterialXShadingNodeImpl<BASE>::updateShader(MHWRender::MShaderInstance& sh
-    for (auto i : inputs)
+    for (const auto& input : inputs)
     {
-        MaterialX::ElementPtr element = document->getDescendant(i.first);
-        MaterialX::ValueElementPtr valueElement = element ? element->asA<MaterialX::ValueElement>() : nullptr;
-        if (valueElement)
+        MaterialX::ElementPtr element = document->getDescendant(input.first);
+        if (!element)
         {
-            std::string inputName(i.second);
-            MString resolvedName(inputName.c_str());
-            const MHWRender::MAttributeParameterMapping* mapping = mappings.findByParameterName(i.second.c_str());
-            if (mapping)
-            {
-                resolvedName = mapping->resolvedParameterName();
-            }
+            continue;
+        }
 
-            if (valueElement->getType() == MaterialX::FILENAME_TYPE_STRING)
-            {
-                // This is the hard-coded OGS convention to associate a texture with a sampler (via post-fix "Sampler" string)
-                std::string textureParameterName(resolvedName.asChar());
+        MaterialX::ValueElementPtr valueElement = element->asA<MaterialX::ValueElement>();
+        if (!valueElement)
+        {
+            continue;
+        }
 
-                // Bind texture and sampler
-                std::string fileName;
-                const std::string& valueString = valueElement->getValueString();
-                if (!valueString.empty())
-                {
-                    MHWRender::MTextureDescription textureDescription;
+        const std::string& inputName = input.second;
+        MString resolvedName(inputName.c_str());
 
-                    // TODO: This should come from the element and not hard-coded.
-                    MHWRender::MSamplerStateDesc samplerDescription;
-                    samplerDescription.filter = MHWRender::MSamplerState::kAnisotropic;
-                    samplerDescription.maxAnisotropy = 16;
+        const MHWRender::MAttributeParameterMapping* mapping = mappings.findByParameterName(input.second.c_str());
+        if (mapping)
+        {
+            resolvedName = mapping->resolvedParameterName();
+        }
 
-                    status = bindFileTexture(shader, textureParameterName, imageSearchPath, valueString,
-                        samplerDescription, textureDescription);
-                }
-            }
+        if (valueElement->getType() == MaterialX::FILENAME_TYPE_STRING)
+        {
+            // This is the hard-coded OGS convention to associate a texture with a sampler (via post-fix "Sampler" string)
+            std::string textureParameterName(resolvedName.asChar());
 
-            // This is unnecessary overhead if we are read-only. Updates should be based on whats
-            // dirty and not everything. There are a lot of attributes on shader graph and this is
-            // a waste of effort currently.
-            if (!_enableEditing)
```

Comment:
Let's remove the editable inputs altogether for now, since they're not part of our milestone? It will be easy to look this code up in history later but they're getting in the way of important changes right now and this is essentially dead code - no one tests it.

Reply:
Agreed.